sklearn/kernel_ridge.py

```diff
@@ -129,6 +133,18 @@ class KernelRidge(MultiOutputMixin, RegressorMixin, BaseEstimator):
     KernelRidge(alpha=1.0)
     """
 
+    _parameter_constraints = {
+        "alpha": [Interval(Real, 0, None, closed="left"), np.ndarray],
+        "kernel": [
+            StrOptions(set(PAIRWISE_KERNEL_FUNCTIONS.keys()) | {"precomputed"}),
+            callable,
+        ],
+        "gamma": [Interval(Real, 0, None, closed="left"), None],
+        "degree": [Interval(Real, 0, None, closed="left")],
+        "coef0": [Interval(Real, 0, None, closed="left")],
+        "kernel_params": "no_validation",
```

Comment:
We should also replace "mapping str to any" from the docstring by `dict`

```suggestion
        "kernel_params": [dict],
```

Reply:
Got it, made the change in docstring as well.